dips/dip-158.md

```diff
@@ -0,0 +1,795 @@
+}
+```
+
+### AuthorizedCommand
+| Functionality |Direction| Description | Command Type | Request Data | Response Data |
+|-|-|-|-|-|-|
+|NOTIFY_AUTHORIZED|Sender > Receiver|This functionality allows the wallet to indicate funds are locked by for future capture. Relevant only for Auth / Capture payments|AuthorizedCommand|Reference ID||
+|
+
+#### AuthorizedCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "AuthorizedCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "AuthorizedCommand",
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98"
+    }  
+}
+
+```
+#### AuthorizedCommand Response
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a"    
+}
+```
+
+### CaptureCommand
+| Functionality |Direction| Description | Command Type | Request Data | Response Data |
+|-|-|-|-|-|-|
+|CAPTURE_PAYMENT|Receiver > Sender|This functionality allows the receiver to capture the locked (authorized) funds. Relevant only for Auth / Capture payments|CaptureCommand|Reference ID||
+|
+
+#### CaptureCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "CaptureCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "CaptureCommand",      
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98"
+  }
+}
+
+```
+#### CaptureCommand Response
+```
+{
+  "_ObjectType": "CommandResponseObject",
+  "status": "success",
+  "cid": "12ce83f6-6d18-0d6e-08b6-c00fdbbf085a"    
+}
+```
+
+### VoidCommand
+| Functionality |Direction| Description | Command Type | Request Data | Response Data |
+|-|-|-|-|-|-|
+|VOID_PAYMENT|Receiver > Sender|This functionality allows the receiver to void the payment. For example, the merchant voided the payment because the purchased products are out of stock. Relevant only for Auth / Capture payments|VoidCommand|Reference ID ; Receiver Status|Sender Status|
+|
+
+#### VoidCommand Request
+Field|Type|Required?|Description|
+|-|-|-|-|
+|reference_id|str|Y|Unique reference ID of this payment. This value should be globaly unique|
+|status|StatusObject|Y|Receiver status. For the void command the status should be abort and the abort code should be void|
+|
+```
+{
+  "_ObjectType": "CommandRequestObject",
+  "command_type": "VoidCommand",
+  "cid": "3185027f-0574-6f55-2668-3a38fdb5de98",
+  "command": {
+    "_ObjectType": "VoidCommand",
+    "reference_id": "4185027f-0574-6f55-2668-3a38fdb5de98",
+    "status": {
+      "status": "abort",
+      "abort_code": "void",
+      "abort_message": "abort message"
+    }
```

Comment:
if the status and abort_code are fixed, perhaps there should only be a string on the reason for the void

Reply:
Since Void is a private case of abort I think void and abort should have a similar structure and make use of the StatusObject.
I believe this consistency will make it easier to implement the protocol and increase clarity.

Reply:
Not sure I fully agree, what happens if I send a void and the abort_code or the status is /not/ correct?

Reply:
Removed PaymentStatusObject from DIP